File: main/cut-the-sticks/cut-the-sticks.cs
using System;
using System.Linq;

internal static class Solution {
    private static void Main()
    {
        var n = int.Parse(Console.ReadLine());
        var a = Console.ReadLine().Split().Select(int.Parse).Take(n).ToArray();
        Array.Sort(a);

        for (var i = 0; i != n; ) {
            Console.WriteLine(n - i);
            var cut = a[i];
            while (i != n && a[i] <= cut) ++i;
            for (var j = i; j != n; ++j) a[j] -= cut;
        }
    }
}

File: main/cut-the-sticks/cut_the_sticks.rb
n = gets.to_i
a = gets.split(' ').map(&:to_i).take(n).sort!

i = 0
while i != n
  puts n - i
  cut = a[i]
  i += 1 while i != n && a[i] <= cut
  (i...n).each { |j| a[j] -= cut }
end
